plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoLoginCallbackProvider.kt

```diff
@@ -18,16 +18,21 @@ import software.aws.toolkits.telemetry.Result
 typealias SsoLoginCallbackProvider = migration.software.aws.toolkits.jetbrains.core.credentials.sso.SsoLoginCallbackProvider
 
 class DefaultSsoLoginCallbackProvider : SsoLoginCallbackProvider {
-    override fun getProvider(ssoUrl: String): SsoLoginCallback = when {
-        JBCefApp.isSupported() -> {
-            if (ssoUrl == SONO_URL) {
-                BearerTokenPromptWithBrowserSupport
-            } else {
-                SsoPromptWithBrowserSupport
-            }
+    override fun getProvider(isAlwaysShowDeviceCode: Boolean, ssoUrl: String): SsoLoginCallback {
+        val deviceCodeProvider = if (ssoUrl == SONO_URL) {
+            DefaultBearerTokenPrompt
+        } else {
+            DefaultSsoPrompt
+        }
+
+        if (isAlwaysShowDeviceCode) {
```

Comment:
nit: could this be included in the when block below?

Reply:
i think it's slightly clearer to the reader what's happening to have it separate

otherwise you'd have to know that the `when {}` conditions are evaluated in order one at a time